### ISSUE TITLE
clean the destination navigator or pop every previous destinations

### ISSUE THREAD

**Issue by getobazooka** [raamcosta/compose-destinations#244]
**How can we pop every previous destination from the destination navigator?**

**For example:**
I have 3 screens/destinations
1- Login Destination
2- Home Destination
3- Detail Destination

When the user logs in, I want him to be unable to go back the login destination using the back button.

**I tried this method, but unfortunately did not work:**

```
navigator.navigate(home_destination) {
    popUpTo(home_destination.route) {
        inclusive = true
    }
}
```

The aforementioned code did not working. But using the navController works.

So basically when the user logs in, how can we prevent the destinationNavigator to go back to login screen using the back button?
Any help would be appreciated.
Thanks.

**Comment by raamcosta** [comment]
```kotlin
navigator.navigate(HomeDestination) {
    popUpTo(LoginDestination) {
        inclusive = true
    }
}
```

This should work. No need to use route either, there is a popUpTo extension function that takes the actual Destination 🙂 

Let me know if this helps!

**Comment by getobazooka** [comment]
Thanks, it worked flawlessly.
But it did not produced the desired effect for this similar scenario: 

HomeDestination -> A_Destination -> B-Destination -> C-Destination

Basically, when the user reaches the C-Destination, I want to clear the whole stack and make the user stay on the C-Destination


navigator.navigate(C-Destination) {
  clearAllPreviousScreens = true
}

Is that possible?

**Comment by raamcosta** [comment]
You can use the same option but popUpTo(NavGraphs.root).

From memory, this should do what you want. 🙂

**Comment by getobazooka** [comment]
Thanks. That's what I was looking for.


```
 fun navigateAndClearPreviousDestinations(destination: Direction) {
            destinationsNavigator.navigate(destination) {
                popUpTo(NavGraphs.root) {
                    inclusive = true
                }
            }
    }
```
